### DISCUSSION TITLE
GPU Issue

### DISCUSSION THREAD

**Discussion by jakecroft** [nextpyp/.github#101] [Q&A] [answered]
Hello, I'm having trouble running jobs that require a GPU through nextPYP.  I have version 0.6.3 on a workstation with two Nvidia RTX 4090s, CUDA version 12.2, driver version 535.146.02.

I've attached the full error, but for example when trying to run aretomo through nextPYP I got :
56	Error: GPU 0 is invalid, skip
58	2024-03-22 11:35:37 [ERROR] pyp/align/core.py:5214 | A GPU must be available for AreTomo2 to run
 
I also can't run motioncor on the GPUs via nextpyp, although I can run it externally. Is there some incompatibility between the CUDA version internal to nextPYPand my GPUs/drivers?

Thanks,
Jake Croft

[gpu_error.txt](https://github.com/nextpyp/.github/files/14794183/gpu_error.txt)

**Comment by abartesaghi** [comment, marked as answer]
@jakecroft, please update to the latest version 0.6.4 and let us know if the error persists. There was an issue with the handling of gpu resources that was fixed in this version.

**Reply by jakecroft** [reply]
Thanks! That seems to have worked.
